linker/Mono.Linker.Steps/ResolveFromAssemblyStep.cs

```diff
@@ -82,8 +82,31 @@ public static void ProcessLibrary (LinkContext context, AssemblyDefinition assem
 			context.Annotations.Push (assembly);
 
 			foreach (TypeDefinition type in assembly.MainModule.Types)
-				MarkType (context, type);
-
+				MarkType (context, type);
+
+			if (assembly.MainModule.HasExportedTypes) {

```

Comment:
Note that only the first part of the block is the same as in MarkStep. I'm happy to factor them into a method but I'm not sure where to put that method. MarkStep and ResolveFromAssemblyStep don't share a common base. Any suggestions?